doc/source/user_guide/integer_na.rst

```diff
@@ -50,17 +49,34 @@ NumPy array.
 You can also pass the list-like object to the :class:`Series` constructor
 with the dtype.
 
-.. ipython:: python
+.. warning::
 
-   s = pd.Series([1, 2, np.nan], dtype="Int64")
-   s
+   Currently :meth:`pandas.array` and :meth:`pandas.Series` use different
```

Comment:
This is my attempt to explain the inconsistency between `pd.array` and `pd.Series` we're introducing. It's not ideal, but I think it's the right behavior for now.

Reply:
Eventually we want to share code between these (and ideally also the Index constructor), right?

Reply:
In the issue (https://github.com/pandas-dev/pandas/issues/29791), I mentioned for that reason the idea of a keyword to control this preference between old or new dtypes. 
(but we can also introduce that at the moment we want to share code if that seems useful then)